README.md
# Waste Storage and Staging Site Selection Tool

The Waste Storage and Staging Tool identifies and ranks potential locations for staging and storing waste. Using a site suitability analysis, the tool considers complicating suitability factors such as soil type, land cover, topography, ease of transportation, and proximity to surface waters. The tool will analyze the various siting criteria for a specified geographic area to identify candidate sites and their total available land surface areas. These criteria can be ranked according to degree of importance and automatically evaluated according to “best fit.” Users can combine the resulting suitability ratings with other important geospatial data to aid in the selection of candidate areas. The tool will allow a user to designate one or more areas within a geographic area of interest, specify attributes and rationale for selection, and save the output for use with other tools or analyses.

Access the [Waste Storage Staging Site Tool User's Guide](/Docs/Final_Waste-Storage-Staging-Tool-Users-Guide_ArcGISPro_02202021.docx) for step-by-step instructions on using the tool.

## Documentation
- [How to contribute to the project](/Docs/CONTRIBUTING.md), learn how to help improve the application. 
- [Changelog](/Docs/CHANGELOG.md), high-level information about the changes in each release.

## Disclaimer

This United States Environmental Protection Agency (EPA) GitHub project code is provided on an "as is" basis and the user assumes responsibility for its use.  EPA has relinquished control of the information and no longer has responsibility to protect the integrity, confidentiality, or availability of the information.  Any reference to specific commercial products, processes, or services by service mark, trademark, manufacturer, or otherwise, does not constitute or imply their endorsement, recommendation or favoring by EPA.  The EPA seal and logo shall not be used in any manner to imply endorsement of any commercial product or activity by EPA or the United States Government. 

[<img src="https://licensebuttons.net/p/mark/1.0/88x31.png">](https://creativecommons.org/publicdomain/zero/1.0/)


Docs/CHANGELOG.md
# Change Log

## 1.0.0 (February 2021)

### Added
- Initial prerelease of new ArcGIS Pro based toolset

### Changed
- Not applicable


debug/commonqa.py
import arcpy,os,sys;
project_root = os.path.dirname(os.path.dirname(os.path.realpath(__file__)));
sys.path.append(project_root);
import util;

###############################################################################
def search_ed():
   
   slope = r"E:\SitingStagingCache\NED13\final\slope_conus_102010.img";
   if not arcpy.Exists(slope):
      slope = r"G:\SitingStagingCache\NED13\final\slope_conus_102010.img";
      
   landcover = r"E:\SitingStagingCache\NLCD2016\final\NLCD_2016_conus_102010.img";
   if not arcpy.Exists(landcover):
      landcover = r"E:\SitingStagingCache\NLCD\final\NLCD_2016_conus_102010.img";
   if not arcpy.Exists(landcover):
      landcover = r"G:\SitingStagingCache\NLCD2016\final\NLCD_2016_conus_102010.img";

   nhd = r"E:\SitingStagingCache\NHD\final\NHD_conus_102010.gdb\NHDPolygons_conus_102010";
   if not arcpy.Exists(nhd):
      nhd = r"E:\SitingStagingCache\NHD\final\NHD_conus.gdb\NHDPolygons_conus_102010";
   if not arcpy.Exists(nhd):
      nhd = r"G:\SitingStagingCache\NHD\final\NHD_conus_102010.gdb\NHDPolygons_conus_102010";

   roads = r"E:\SitingStagingCache\USGSTNM\final\TNMT_conus_102010.gdb\ROADS_conus_201020";
   if not arcpy.Exists(roads):
      roads = r"G:\SitingStagingCache\USGSTNM\final\TNMT_conus_102010.gdb\ROADS_conus_201020";

   ssurgo = r"E:\SitingStagingCache\EsriSSURGO\final\EsriSSURGO_conus_102010.gdb\EsriSSURGO_conus_102010";
   if not arcpy.Exists(ssurgo):
      ssurgo = r"G:\SitingStagingCache\EsriSSURGO\final\EsriSSURGO_conus_102010.gdb\EsriSSURGO_conus_102010";
      
   return {
       'slope'    : slope
      ,'landcover': landcover
      ,'nhd'      : nhd
      ,'roads'    : roads
      ,'ssurgo'   : ssurgo
   };
   
###############################################################################
def coords2fc(
    coordinates
   ,flds
   ,srid        = None
   ,temp_name   = None
):

   if temp_name is None:
      temp_name = "temp_aoi";
      
   if srid is None:
      sr = arcpy.SpatialReference(4269);  
   else:
      sr = arcpy.SpatialReference(srid);
      
   if arcpy.Exists(arcpy.env.scratchWorkspace + os.sep + temp_name):
      arcpy.Delete_management(arcpy.env.scratchWorkspace + os.sep + temp_name);
      
   arcpy.CreateFeatureclass_management(
       out_path          = arcpy.env.scratchWorkspace
      ,out_name          = temp_name
      ,geometry_type     = "POLYGON"
      ,has_m             = "DISABLED"
      ,has_z             = "DISABLED"
      ,spatial_reference = sr
      ,config_keyword    = None
   );
   
   if flds is not None:
      arcpy.management.AddFields(
          arcpy.env.scratchWorkspace + os.sep + temp_name
         ,[flds]
      );
   
   return temp_name;
   
###############################################################################
def temp_aoi(
    coordinates
   ,srid        = None
   ,temp_name   = None
   ,aoi_name    = None
):

   if aoi_name is None:
      aoi_name = 'aoi';
      
   temp_name = coords2fc(
       coordinates = coordinates
      ,flds        = ['aoi_name','TEXT','Area of Interest Name' ,255, None,'']
      ,srid        = srid
      ,temp_name   = temp_name
   );
   
   with arcpy.da.InsertCursor(
       arcpy.env.scratchWorkspace + os.sep + temp_name
      ,['aoi_name','SHAPE@']
   ) as cursor:
      cursor.insertRow([aoi_name,coordinates]);

   return arcpy.env.scratchWorkspace + os.sep + temp_name;
   
###############################################################################
def make_selection_by_attr(
    where_clause
   ,input_dataset
   ,lyr_name
):

   if arcpy.Exists(lyr_name):
      arcpy.Delete_management(lyr_name);
   
   arcpy.MakeFeatureLayer_management(
       in_features = input_dataset
      ,out_layer   = lyr_name
   );

   rez = arcpy.SelectLayerByAttribute_management(
       in_layer_or_view = lyr_name
      ,selection_type   = 'NEW_SELECTION'
      ,where_clause     = where_clause
   );
   
   return lyr_name;
   
###############################################################################
def make_selection_by_coords(
    coordinates
   ,input_dataset
   ,lyr_name
   ,srid         = None
):

   slicer = temp_aoi(
       coordinates = coordinates
      ,srid        = srid
      ,temp_name   = "slicer"
   );
   
   if arcpy.Exists(lyr_name):
      arcpy.Delete_management(lyr_name);
   
   arcpy.MakeFeatureLayer_management(
       in_features = input_dataset
      ,out_layer   = lyr_name
   );

   rez = arcpy.SelectLayerByLocation_management(
       in_layer         = lyr_name
      ,overlap_type     = 'INTERSECT'
      ,select_features  = slicer
      ,selection_type   = 'NEW_SELECTION'
   );
   
   return lyr_name;
   
   

debug/qa_newmunster.py
import arcpy,os,sys;
from importlib.util import spec_from_loader, module_from_spec;
from importlib.machinery import SourceFileLoader;
import commonqa;

if arcpy.CheckExtension("Spatial") == "Available":
   arcpy.CheckOutExtension("Spatial");
else:
   raise Exception('Spatial Analysis Extension not available.');
print("");

###############################################################################
# Step 10
# Add project path to python path and import util forcing project to aprx file
###############################################################################
project_root = os.path.dirname(os.path.dirname(os.path.realpath(__file__)));
print("Step 10: Importing utilities from project at \n   " + project_root + os.sep + "util.py");
sys.path.append(project_root);
import util;
util.g_prj = util.g_pn + os.sep + "EPA-Storage-Staging-Site-Tool.aprx";

###############################################################################
# Step 20
# Short circuit the toolbox and load the util toolbox directly in order to 
# access the parameters.  ArcPy remains a buggy mess and these contortions are
# the only way I can test the tools without just recreating steps from scratch
###############################################################################
toolbx = os.path.join(util.g_pn,"EPA-Storage-Staging-Site-Util.pyt");
print("Step 20: Sideloading util toolbox from \n   " + toolbx);
spec = spec_from_loader(
    'util_sideloaded'
   ,SourceFileLoader(
       'util_sideloaded'
      ,toolbx
   )
);
module = module_from_spec(spec);
spec.loader.exec_module(module);
sys.modules['util_sideloaded'] = module;
import util_sideloaded;
util.g_prj = util.g_pn + os.sep + "EPA-Storage-Staging-Site-Tool.aprx";

###############################################################################
# Step 30
# Short circuit the toolbox and load the tool toolbox
###############################################################################
toolbx = os.path.join(util.g_pn,"EPA-Storage-Staging-Site-Tool.pyt");
print("Step 30: Sideloading tool toolbox from \n   " + toolbx);
spec = spec_from_loader(
    'tool_sideloaded'
   ,SourceFileLoader(
       'tool_sideloaded'
      ,toolbx
   )
);
module = module_from_spec(spec);
spec.loader.exec_module(module);
sys.modules['tool_sideloaded'] = module;
import tool_sideloaded;
util.g_prj = util.g_pn + os.sep + "EPA-Storage-Staging-Site-Tool.aprx";

###############################################################################
# Step 40
# Set project common variables
###############################################################################
print("Step 40: Loading workspace and project.")
wrksp = os.path.join(util.g_pn,"EPA-Storage-Staging-Site-Tool.gdb");
arcpy.env.workspace = wrksp;
print("   workspace: " + str(wrksp));
arcpy.env.scratchWorkspace = os.path.join(util.g_pn,"scratch.gdb");
arcpy.env.overwriteOutput = True;
aprx = arcpy.mp.ArcGISProject(util.g_prj);
default_map = aprx.listMaps('*')[0];
print("   project: " + str(aprx.filePath));

if not arcpy.Exists(aprx.defaultGeodatabase):
   arcpy.CreateFileGDB_management(
       out_folder_path = util.g_pn
      ,out_name        = "EPA-Storage-Staging-Site-Tool.gdb"
   );

###############################################################################
# Step 50
# Clear all scenarios
###############################################################################
print("Step 50: Scenario Setup."); 
st = util_sideloaded.ScenarioSetup();
parameters = st.getParameterInfo();
parameters[1].value = False;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 60
# Add new Scenario if not preexisting
###############################################################################
if not util.scenarioIDExists(
    id   = 'NewMunster1'
   ,aprx = None
):
   print("Step 60: Add new scenario."); 
   st = util_sideloaded.LoadNewScenario();
   parameters = st.getParameterInfo();
   parameters[1].value  = 'NewMunster1';
   parameters[2].value  = 'NewMunster';
   parameters[3].value  = True;
   parameters[4].value  = True;
   parameters[5].value  = True;
   parameters[6].value  = util.g_default_nhdgridsize;
   parameters[7].value  = True;
   parameters[8].value  = util.g_default_tnmroadsgridsize;
   parameters[9].value  = True;
   parameters[10].value = util.g_default_ssurgogridsize;
   parameters[11].value = 'test scenario';
   messages = None;
   rez = st.execute(parameters,messages);

###############################################################################
# Step 70
# Run ConfirmSuitabilityCriteria
###############################################################################
print("Step 70: Run ConfirmSuitabilityCriteria.");
st = tool_sideloaded.ConfirmSuitabilityCriteria();
parameters = st.getParameterInfo();
parameters[2].value  = 'NewMunster1';
parameters[3].value  = default_map.name;
parameters[4].value  = "Value > 10";
parameters[5].value  = "Value = 11 OR Value = 23 OR Value = 24 OR Value = 41 OR Value = 42 OR Value = 43 OR Value = 81 OR Value = 82 OR Value = 90 OR Value = 95";
parameters[6].value  = "Value <= 500";
parameters[7].value  = "Value < 200 OR Value > 500";
parameters[8].value  = "hydgrpdcd = 'B' OR hydgrpdcd = 'B/D' OR hydgrpdcd = 'A' OR hydgrpdcd = 'A/D' OR hydgrpdcd = ' '";
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 80
# Run SpecifyCriteriaWeight
###############################################################################
print("Step 80: Run SpecifyCriteriaWeight."); 
st = tool_sideloaded.SpecifyCriteriaWeight();
parameters = st.getParameterInfo();
parameters[2].value  = 'NewMunster1';
parameters[3].value  = default_map.name;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 90
# Fake a layer selection
###############################################################################
print("Step 90: Make selection using a bounding box.");
coordinates = [
    (-88.163192,42.600514)
   ,(-88.118496,42.586283)
   ,(-88.127011,42.529928)
   ,(-88.190187,42.534740)
   ,(-88.163192,42.600514)
];

lyr_newmunster = commonqa.make_selection_by_coords(
    coordinates   = coordinates
   ,input_dataset = 'NewMunster1_WEIGHTEDSUM'
   ,lyr_name      = 'NewMunster1_lyrtmp'
   ,srid          = 4269
);

###############################################################################
# Step 100
# Run FinalizeStagingParcelSelection
###############################################################################
print("Step 100: Run FinalizeStagingParcelSelection."); 
st = tool_sideloaded.FinalizeStagingParcelSelection();
parameters = st.getParameterInfo();
parameters[2].value  = 'NewMunster1';
parameters[3].value  = default_map.name;
parameters[4].value  = lyr_newmunster;
parameters[5].value  = True;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 110
# Run ExportSaveResults
###############################################################################
if arcpy.Exists(arcpy.env.scratchFolder + os.sep + 'NewMunster1.xlsx'):
   arcpy.Delete_management(arcpy.env.scratchFolder + os.sep + 'NewMunster1.xlsx');
   
print("Step 210: Run ExportSaveResults."); 
st = tool_sideloaded.ExportSaveResults();
parameters = st.getParameterInfo();
parameters[2].value  = 'NewMunster1';
parameters[3].value  = 'Disabled';
parameters[6].value  = arcpy.env.scratchFolder + os.sep + 'NewMunster1.xlsx'
messages = None;
rez = st.execute(parameters,messages);

print("\nqa_newmunster complete."); 



ArcMap/Script/Reclassify.py
# Name: 2 Reclassify
# Description: Reclassify rasters as part of the Staging Site Selection Model
# Requirements: Spatial Analyst Extension

# Import system modules
import arcpy, os
from arcpy import env
from arcpy.sa import *

arcpy.env.overwriteOutput = True

# Define root directory and define geodatabase name
folder = os.path.dirname(arcpy.mapping.MapDocument('CURRENT').filePath)
Tooldata = os.path.join(folder, "Tooldata")
geodatabase = os.path.join(Tooldata, "ModelOutputTables.gdb")

# Set environment settings
#env.workspace = ""

# Set local variables
inSlope = Raster(os.path.join(geodatabase, "Extract_Slope"))
inRoads = Raster(os.path.join(geodatabase, "Extract_Roads"))
inSSURGO = Raster(os.path.join(geodatabase, "Extract_SSURGO"))
inNHD = Raster(os.path.join(geodatabase, "Extract_NHD"))
inLAND = Raster(os.path.join(geodatabase, "Extract_LAND"))
inTrueRaster = 0
inFalseConstant = 1

#SQL statements
#whereClause = "Value > 10" #Slope
#whereClause = "Value < 200 OR Value > 500" #Roads
#whereClause = "hydgrpdcd = 'B' OR hydgrpdcd = 'B/D' OR hydgrpdcd = 'A' OR hydgrpdcd = 'A/D' OR hydgrpdcd = ' '" #SSURGO
#whereClause = "Value <= 500" #NHD
#whereClause = "Value = 11 OR Value = 23 OR Value = 24 OR Value = 41 OR Value = 42 OR Value = 43 OR Value = 81 OR Value = 82 OR Value = 90 OR Value = 95" #Land

# Check out the ArcGIS Spatial Analyst extension license
arcpy.CheckOutExtension("Spatial")

# Execute Con
arcpy.AddMessage("Processing Slope...")
outCon = Con(inSlope, inTrueRaster, inFalseConstant, sys.argv[2])
arcpy.AddMessage("Processing Roads...")
outCon2 = Con(inRoads, inTrueRaster, inFalseConstant, sys.argv[1])
arcpy.AddMessage("Processing Soils...")
outCon3 = Con(inSSURGO, inTrueRaster, inFalseConstant, sys.argv[4])
arcpy.AddMessage("Processing Surface Water...")
outCon4 = Con(inNHD, inTrueRaster, inFalseConstant, sys.argv[5])
arcpy.AddMessage("Processing Land Classification...")
outCon5 = Con(inLAND, inTrueRaster, inFalseConstant, sys.argv[3])

arcpy.AddMessage("Saving outputs...")

# Save the outputs 
outCon.save(os.path.join(geodatabase, "Reclass_Slope"))
outCon2.save(os.path.join(geodatabase, "Reclass_Roads"))
outCon3.save(os.path.join(geodatabase, "Reclass_SSURGO"))
outCon4.save(os.path.join(geodatabase, "Reclass_NHD"))
outCon5.save(os.path.join(geodatabase, "Reclass_LAND"))

arcpy.AddMessage("Done!")


debug/qa_hermith.py
import arcpy,os,sys;
from importlib.util import spec_from_loader, module_from_spec;
from importlib.machinery import SourceFileLoader;
import commonqa;

if arcpy.CheckExtension("Spatial") == "Available":
   arcpy.CheckOutExtension("Spatial");
else:
   raise Exception('Spatial Analysis Extension not available.');
print("");

###############################################################################
# Step 10
# Add project path to python path and import util forcing project to aprx file
###############################################################################
project_root = os.path.dirname(os.path.dirname(os.path.realpath(__file__)));
print("Step 10: Importing utilities from project at \n   " + project_root + os.sep + "util.py");
sys.path.append(project_root);
import util;
util.g_prj = util.g_pn + os.sep + "EPA-Storage-Staging-Site-Tool.aprx";

###############################################################################
# Step 20
# Locate the Essential Datasets
###############################################################################
ed = commonqa.search_ed();
  
if not arcpy.Exists(ed['slope']):
   raise Exception("Essential Data Slope input not found.");
   
if not arcpy.Exists(ed['landcover']):
   raise Exception("Essential Data Landcover input not found.");
   
if not arcpy.Exists(ed['nhd']):
   raise Exception("Essential Data NHD input not found.");
   
if not arcpy.Exists(ed['roads']):
   raise Exception("Essential Data Roads input not found.");
   
if not arcpy.Exists(ed['ssurgo']):
   raise Exception("Essential Data SSurgo input not found.");

print("Step 20: Found all ED components");

###############################################################################
# Step 30
# Short circuit the toolbox and load the util toolbox directly in order to 
# access the parameters.  ArcPy remains a buggy mess and these contortions are
# the only way I can test the tools without just recreating steps from scratch
###############################################################################
toolbx = os.path.join(util.g_pn,"EPA-Storage-Staging-Site-Util.pyt");
print("Step 30: Sideloading util toolbox from \n   " + toolbx);
spec = spec_from_loader(
    'util_sideloaded'
   ,SourceFileLoader(
       'util_sideloaded'
      ,toolbx
   )
);
module = module_from_spec(spec);
spec.loader.exec_module(module);
sys.modules['util_sideloaded'] = module;
import util_sideloaded;
util.g_prj = util.g_pn + os.sep + "EPA-Storage-Staging-Site-Tool.aprx";

###############################################################################
# Step 40
# Short circuit the toolbox and load the tool toolbox
###############################################################################
toolbx = os.path.join(util.g_pn,"EPA-Storage-Staging-Site-Tool.pyt");
print("Step 40: Sideloading tool toolbox from \n   " + toolbx);
spec = spec_from_loader(
    'tool_sideloaded'
   ,SourceFileLoader(
       'tool_sideloaded'
      ,toolbx
   )
);
module = module_from_spec(spec);
spec.loader.exec_module(module);
sys.modules['tool_sideloaded'] = module;
import tool_sideloaded;
util.g_prj = util.g_pn + os.sep + "EPA-Storage-Staging-Site-Tool.aprx";

###############################################################################
# Step 50
# Set project common variables
###############################################################################
print("Step 50: Loading workspace and project.")
wrksp = os.path.join(util.g_pn,"EPA-Storage-Staging-Site-Tool.gdb");
arcpy.env.workspace = wrksp;
print("   workspace: " + str(wrksp));
arcpy.env.scratchWorkspace = os.path.join(util.g_pn,"scratch.gdb");
arcpy.env.overwriteOutput = True;
aprx = arcpy.mp.ArcGISProject(util.g_prj);
print("   project: " + str(aprx.filePath));

if not arcpy.Exists(aprx.defaultGeodatabase):
   arcpy.CreateFileGDB_management(
       out_folder_path = util.g_pn
      ,out_name        = "EPA-Storage-Staging-Site-Tool.gdb"
   );

###############################################################################
# Step 60
# Clear all AOIs
###############################################################################
print("Step 60: Execute AOI Setup."); 
st = util_sideloaded.AOISetup();
parameters = st.getParameterInfo();
parameters[1].value = True;
parameters[2].value = True;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 70
# Create Area of Interest fc
###############################################################################
print("Step 70: Execute AOI Creation.");
coordinates = [
    (-90.56545257568358,43.18302464679583)
   ,(-90.43052673339844,43.17914423586488)
   ,(-90.41885375976562,43.224191874050916)
   ,(-90.43069839477539,43.282704074896195)
   ,(-90.5958366394043,43.277330299293055)
   ,(-90.59429168701172,43.18527767545014)
   ,(-90.56545257568358,43.18302464679583)
];

temp_aoi = commonqa.temp_aoi(
    coordinates = coordinates
   ,temp_name   = "temp_aoi"
);

###############################################################################
# Step 80
# Add new Area of Interest
###############################################################################
print("Step 80: Add new Area of Interest."); 
st = util_sideloaded.LoadNewAOI();
parameters = st.getParameterInfo();
parameters[1].value  = 'EagleCave';
parameters[2].value  = None;
parameters[3].value  = temp_aoi;
parameters[5].value  = ed['slope'];
parameters[6].value  = ed['landcover'];
parameters[7].value  = ed['nhd'];
parameters[8].value  = ed['roads'];
parameters[9].value  = ed['ssurgo'];
parameters[10].value = 5;
parameters[11].value = 'test aoi, please delete';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 90
# Rename Area of Interest
###############################################################################
print("Step 90: Rename Area of Interest."); 
st = util_sideloaded.RenameAOI();
parameters = st.getParameterInfo();
parameters[1].value = 'EagleCave';
parameters[2].value = 'HermitHole';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 100
# Delete Area of Interest
###############################################################################
print("Step 100: Delete Area of Interest."); 
st = util_sideloaded.DeleteAOI();
parameters = st.getParameterInfo();
parameters[1].value = 'HermitHole';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 110
# Add new Area of Interest
###############################################################################
print("Step 110: Read Area of Interest."); 
st = util_sideloaded.LoadNewAOI();
parameters = st.getParameterInfo();
parameters[1].value  = 'EagleCave';
parameters[2].value  = None;
parameters[3].value  = temp_aoi;
parameters[5].value  = ed['slope'];
parameters[6].value  = ed['landcover'];
parameters[7].value  = ed['nhd'];
parameters[8].value  = ed['roads'];
parameters[9].value  = ed['ssurgo'];
parameters[10].value = 5;
parameters[11].value = 'test aoi, please delete';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 120
# Clear all scenarios
###############################################################################
print("Step 120: Scenario Setup."); 
st = util_sideloaded.ScenarioSetup();
parameters = st.getParameterInfo();
parameters[1].value = True;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 130
# Add new Scenario
###############################################################################
print("Step 130: Add new scenario."); 
st = util_sideloaded.LoadNewScenario();
parameters = st.getParameterInfo();
parameters[1].value  = 'EagleCave1';
parameters[2].value  = 'EagleCave';
parameters[3].value  = True;
parameters[4].value  = True;
parameters[5].value  = True;
parameters[6].value  = util.g_default_nhdgridsize;
parameters[7].value  = True;
parameters[8].value  = util.g_default_tnmroadsgridsize;
parameters[9].value  = True;
parameters[10].value = util.g_default_ssurgogridsize;
parameters[11].value = 'test scenario, please delete';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 140
# Duplicate Scenario
###############################################################################
print("Step 140: Duplicate scenario."); 
st = util_sideloaded.DuplicateScenario();
parameters = st.getParameterInfo();
parameters[1].value  = 'EagleCave1';
parameters[2].value  = 'EagleCave99';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 150
# Rename Scenario
###############################################################################
print("Step 150: Rename scenario."); 
st = util_sideloaded.RenameScenario();
parameters = st.getParameterInfo();
parameters[1].value  = 'EagleCave99';
parameters[2].value  = 'HermitHole';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 160
# Delete Scenario
###############################################################################
print("Step 160: Delete scenario."); 
st = util_sideloaded.DeleteScenario();
parameters = st.getParameterInfo();
parameters[1].value  = 'HermitHole';
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 170
# Run ConfirmSuitabilityCriteria
###############################################################################
print("Step 170: Run ConfirmSuitabilityCriteria.");
default_map = aprx.listMaps('*')[0];
st = tool_sideloaded.ConfirmSuitabilityCriteria();
parameters = st.getParameterInfo();
parameters[2].value  = 'EagleCave1';
parameters[3].value  = default_map.name;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 180
# Run SpecifyCriteriaWeight
###############################################################################
print("Step 180: Run SpecifyCriteriaWeight."); 
st = tool_sideloaded.SpecifyCriteriaWeight();
parameters = st.getParameterInfo();
parameters[2].value  = 'EagleCave1';
parameters[3].value  = default_map.name;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 190
# Fake a layer selection
###############################################################################
print("Step 190: Make selection against weighted sum.");
lyr_eaglecave = commonqa.make_selection_by_attr(
    where_clause = '"suitability_score" = 1'
   ,input_dataset = 'EagleCave1_WEIGHTEDSUM'
   ,lyr_name      = 'EagleCave1_lyrtmp'
);

###############################################################################
# Step 200
# Run FinalizeStagingParcelSelection
###############################################################################
print("Step 200: Run FinalizeStagingParcelSelection."); 
st = tool_sideloaded.FinalizeStagingParcelSelection();
parameters = st.getParameterInfo();
parameters[2].value  = 'EagleCave1';
parameters[3].value  = default_map.name;
parameters[4].value  = lyr_eaglecave;
parameters[5].value  = True;
messages = None;
rez = st.execute(parameters,messages);

###############################################################################
# Step 210
# Run ExportSaveResults
###############################################################################
if arcpy.Exists(arcpy.env.scratchFolder + os.sep + 'EagleCave1.xlsx'):
   arcpy.Delete_management(arcpy.env.scratchFolder + os.sep + 'EagleCave1.xlsx');
   
print("Step 210: Run ExportSaveResults."); 
st = tool_sideloaded.ExportSaveResults();
parameters = st.getParameterInfo();
parameters[2].value  = 'EagleCave1';
parameters[3].value  = 'Disabled';
parameters[6].value  = arcpy.env.scratchFolder + os.sep + 'EagleCave1.xlsx'
messages = None;
rez = st.execute(parameters,messages);

print("\nqa_hermith complete."); 

